documentatie/flowcharts/date-entry-page.md

```diff
@@ -17,55 +17,92 @@ flowchart TD
-    error-cur-page{error-cur-page?}
+    error-any-next-page{error-any-next-page?}
+    error-any-prev-page{error-any-prev-page?}
 
-    warning-next-page{warning-next-page?}
-    warning-prev-page{warning-prev-page?}
-    warning-cur-page{warning-cur-page?}
+    warning-any-next-page{warning-any-next-page?}
+    warning-any-prev-page{warning-any-prev-page?}
 
-    cur-page-submitted{cur-page-submitted?}
+    already-has-error-icon{already-has-error-icon?}
 
     fill-cached-input
 
     %% flow
-    flow-start --> error-prev-page
+    flow-start --> current-icon-cur-page
+    
+    current-icon-cur-page --> error-any-prev-page
+
+    error-any-prev-page -- yes --> illegal-state
+    error-any-prev-page -- no --> error-any-next-page
+
+    error-any-next-page -- yes --> error-icon-next-page
+    error-icon-next-page --> warning-any-prev-page
+    error-any-next-page -- no --> warning-any-prev-page
+
+    warning-any-prev-page -- no --> warning-any-next-page
+    warning-any-prev-page -- yes --> already-has-error-icon
+    already-has-error-icon -- yes --> warning-any-next-page
+    already-has-error-icon -- no --> warning-icon-prev-page
+    warning-icon-prev-page --> warning-any-next-page
+
+    warning-any-next-page -- no --> fill-cached-input
+    warning-any-next-page -- yes --> warning-icon-next-page
+    warning-icon-next-page --> fill-cached-input
+
+    fill-cached-input --> flow-end
+
+```
+
+
+## Render form
+
+Render happens based on last received API response.
 
-    error-prev-page -- yes --> illegal-state
-    error-prev-page -- no --> error-next-page
+```mermaid
+flowchart TD
+
+    %% elements
+    flow-start([start])
+    flow-end([end])
+    illegal-state{{illegal state}}
+
+    show-error
+    show-warning
 
-    error-next-page -- yes --> error-icon-next-page
-    error-icon-next-page --> error-cur-page
-    error-next-page -- no --> error-cur-page
+    error-any-prev-page{error-any-prev-page?}
+    error-cur-page{error-cur-page?}
 
-    error-cur-page -- yes --> cur-page-submitted
-    cur-page-submitted -- yes --> error-icon-cur-page
-    error-icon-cur-page --> show-error
-    cur-page-submitted -- no --> warning-prev-page
-    error-cur-page -- no --> warning-prev-page
+    warning-cur-page{warning-cur-page?}
 
-    warning-prev-page -- no --> warning-next-page
-    warning-prev-page -- yes --> warning-icon-prev-page
-    warning-icon-prev-page --> warning-next-page
+    cur-page-submitted-error{cur-page-submitted?}
+    cur-page-submitted-warning{cur-page-submitted?}
 
-    warning-next-page -- no --> warning-cur-page
-    warning-next-page -- yes --> warning-icon-next-page
-    warning-icon-next-page --> warning-cur-page
+    fill-cached-input
+
+    %% flow
+    flow-start --> error-any-prev-page
+
+    error-any-prev-page -- yes --> illegal-state
+    error-any-prev-page -- no -->  error-cur-page
+
+    error-cur-page -- yes --> cur-page-submitted-error
+    cur-page-submitted-error -- yes --> show-error
```

Comment:
I think this diagram can be simplified by putting `cur-page-submitted` above `error-cur-page`, because then you won't need to repeat `cur-page-submitted`. If no: you just render the cashed data. If yes, first check for errors, then check for warnings.

Reply:
good catch! updated the diagram, also removing the "illegal state" part in favor of a comment at the top of the document